h/services/bulk_annotation.py

```diff
@@ -0,0 +1,190 @@
+from typing import List, Union
+
+import sqlalchemy as sa
+from sqlalchemy.engine import Row
+from sqlalchemy.orm import Session
+from sqlalchemy.sql import Select
+
+from h.models import Annotation, AnnotationModeration, Group, User
+
+
+class BadDateFilter(ValueError):
+    """There is something wrong with the date filter provided."""
+
+
+class BadFieldSpec(ValueError):
+    """There is something wrong with the field you have specified."""
```

Comment:
I did this because they are semantically something wrong with a value.

Reply:
I'd avoid subclassing `ValueError` just because it means code could unintentionally catch these exceptions when trying to catch some other `ValueError`, and I don't think the semanticneess of `ValueError` has any practical benefit